e2e/cypress/integration/interactive_button/interactive_button_spec.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/**
+* Note: This test requires webhook server running. Initiate `npm run start:webhook` to start.
+*/
+
+import {getMessageButtonsPayload} from '../../utils';
+
+const payload = getMessageButtonsPayload();
+
+let incomingWebhook;
+
+describe('Interactive Message: Button', () => {
+    before(() => {
+        // Set required ServiceSettings
+        const newSettings = {
+            ServiceSettings: {
+                AllowedUntrustedInternalConnections: 'localhost',
+                EnablePostUsernameOverride: true,
+                EnablePostIconOverride: true,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Login as sysadmin and ensure that teammate name display setting is set to default 'username'
+        cy.apiLogin('sysadmin');
+        cy.apiSaveTeammateNameDisplayPreference('username');
+        cy.apiSaveMessageDisplayPreference('clean');
+
+        // # Visit '/' and create incoming webhook
+        cy.visit('/ad-1/channels/town-square');
+        cy.getCurrentChannelId().then((id) => {
+            const newIncomingHook = {
+                channel_id: id,
+                channel_locked: true,
+                description: 'Incoming webhook interactive menu',
+                display_name: 'menuIn' + Date.now(),
+            };
+
+            cy.apiCreateWebhook(newIncomingHook).then((hook) => {
+                incomingWebhook = hook;
+            });
+        });
+    });
+
+    it('IM10164 - Include extra metadata when clicking an interactive button', () => {
+        // # Post an incoming webhook
+        cy.postIncomingWebhook({url: incomingWebhook.url, data: payload});
+
+        // # Get message attachment from the last post
+        let interactiveButtonPostId;
+        cy.getLastPostId().then((postId) => {
+            interactiveButtonPostId = postId;
+            cy.get(`#messageAttachmentList_${postId}`).as('messageAttachmentList');
+        });
+
+        // * Verify each element of message attachment list and then click "Ephemeral" button
+        cy.get('@messageAttachmentList').within(() => {
+            cy.get('.attachment__thumb-pretext').should('be.visible').and('have.text', 'This is the attachment pretext.');
+            cy.get('.post-message__text-container').should('be.visible').and('have.text', 'This is the attachment text for Interactive Message Buttons.');
+            cy.findByText('Ephemeral Message').should('be.visible').click();
+        }).then(() => {
+            // * Verify metadata on click to "Ephemeral" button
```

Comment:
Is clicking the "Ephemeral Message" button supposed to do anything? Right now, nothing happens when the button is clicked, and the test for me is failing right after step in verifyMessage (the last post message cannot be found so I'm wondering if this was the message that the Ephemeral Message button was supposed to generate):
`CypressError: Timed out retrying: expected '<div#postMessageText_d4k857ue5pyemfyzfxr6dmr3we.post-message__text>' to be 'visible'`